src/Microsoft.IdentityModel.JsonWebTokens/JsonWebTokenHandler.cs

```diff
@@ -755,13 +755,14 @@ private static string EncryptTokenPrivate(string innerJwt, EncryptingCredentials
                 throw LogHelper.LogExceptionMessage(new ArgumentException(TokenLogMessages.IDX10620));
 
```

Comment:
This is a copy from the JwtSecurityTokenHandler.EncryptToken() but missing a null check on encryptingCredentials.
Suggest adding the GetCryptoProviderFactory() to the base class TokenHandler:

        protected static CryptoProviderFactory GetCryptoProviderFactory(EncryptingCredentials encryptingCredentials)
        {
            if (encryptingCredentials == null)
                throw LogHelper.LogArgumentNullException(nameof(encryptingCredentials));

            var cryptoProviderFactory = encryptingCredentials.CryptoProviderFactory ?? encryptingCredentials.Key.CryptoProviderFactory;

            if (cryptoProviderFactory == null)
                throw LogHelper.LogExceptionMessage(new ArgumentException(TokenLogMessages.IDX10620));

            return cryptoProviderFactory;
        }

And both JsonWebTokenHandler and JwtSecurityTokenHandler get the factory in a consistent manner:

            var cryptoProviderFactory = GetCryptoProviderFactory(encryptingCredentials);

Reply:
After inspecting the calls into these two methods, I've realized that in neither case it is possible for these two methods to receive encyptingCredentials with a null value (each code path has a previous check for null value. As these two methods are private, I'll remove the check for null value in JwtSecurityTokenHandler.EncryptToken()